src/defineFunction.js

```diff
@@ -3,92 +3,101 @@ import functions from "./functions";
     names: Array<string>,
 
     // Properties that control how the functions are parsed.
-    props: {
-        // The number of arguments the function takes.
-        numArgs?: number,
-
-        // An array corresponding to each argument of the function, giving the
-        // type of argument that should be parsed. Its length should be equal
-        // to `numArgs + numOptionalArgs`.
-        argTypes?: ArgType[],
-
-        // The greediness of the function to use ungrouped arguments.
-        //
-        // E.g. if you have an expression
-        //   \sqrt \frac 1 2
-        // since \frac has greediness=2 vs \sqrt's greediness=1, \frac
-        // will use the two arguments '1' and '2' as its two arguments,
-        // then that whole function will be used as the argument to
-        // \sqrt. On the other hand, the expressions
-        //   \frac \frac 1 2 3
-        // and
-        //   \frac \sqrt 1 2
-        // will fail because \frac and \frac have equal greediness
-        // and \sqrt has a lower greediness than \frac respectively. To
-        // make these parse, we would have to change them to:
-        //   \frac {\frac 1 2} 3
-        // and
-        //   \frac {\sqrt 1} 2
-        //
-        // The default value is `1`
-        greediness?: number,
-
-        // Whether or not the function is allowed inside text mode
-        // (default false)
-        allowedInText?: boolean,
-
-        // Whether or not the function is allowed inside text mode
-        // (default true)
-        allowedInMath?: boolean,
-
-        // (optional) The number of optional arguments the function
-        // should parse. If the optional arguments aren't found,
-        // `null` will be passed to the handler in their place.
-        // (default 0)
-        numOptionalArgs?: number,
-
-        // Must be true if the function is an infix operator.
-        infix?: boolean,
-    },
+    props: FunctionPropSpec,
 
     // The handler is called to handle these functions and their arguments.
-    // It receives two arguments:
-    //   - context contains information and references provided by the parser
-    //   - args is an array of arguments obtained from TeX input
-    // The context contains the following properties:
-    //   - funcName: the text (i.e. name) of the function, including \
-    //   - parser: the parser object
-    //   - lexer: the lexer object
-    // The latter three should only be used to produce error messages.
     //
     // The function should return an object with the following keys:
     //   - type: The type of element that this is. This is then used in
     //          buildHTML/buildMathML to determine which function
     //          should be called to build this node into a DOM node
     // Any other data can be added to the object, which will be passed
     // in to the function in buildHTML/buildMathML as `group.value`.
-    handler: (context: any, args: any) => T,
```

Comment:
Note that the output value of `handler` is actually unrelated to the `group` parameter passed into `htmlBuilder`.